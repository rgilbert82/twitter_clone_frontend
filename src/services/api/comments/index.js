export { default as getCommentsAPI } from './getComments.js';
export { default as createCommentAPI } from './createComment.js';
export { default as editCommentAPI } from './editComment.js';
export { default as deleteCommentAPI } from './deleteComment.js';
