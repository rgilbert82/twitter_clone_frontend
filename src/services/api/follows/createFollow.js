import { get_API_URL } from '../../misc';

export default (user_id) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/follows`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          resolve(request.responseText);
        }
      }
    }

    request.open('POST', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.setRequestHeader('Content-Type', 'application/json');
    request.send(JSON.stringify({ user_id: user_id }));
  });
}
