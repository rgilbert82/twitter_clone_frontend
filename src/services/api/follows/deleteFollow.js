import { get_API_URL } from '../../misc';

export default (user_id) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/follows/${user_id}`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    }

    request.open('DELETE', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.send();
  });
}
