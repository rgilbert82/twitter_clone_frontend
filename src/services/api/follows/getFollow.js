import { get_API_URL } from '../../misc';

export default (userID) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/follows/${userID}`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          resolve(request.responseText);
        }
      }
    }

    request.open('GET', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.send();
  });
}
