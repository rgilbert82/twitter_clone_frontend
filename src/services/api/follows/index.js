export { default as getFollowAPI } from './getFollow.js';
export { default as createFollowAPI } from './createFollow.js';
export { default as deleteFollowAPI } from './deleteFollow.js';
export { default as getUserFollowersAPI } from './getUserFollowers.js';
export { default as getUserFollowingsAPI } from './getUserFollowings.js';
