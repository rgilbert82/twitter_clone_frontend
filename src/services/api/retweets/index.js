export { default as createRetweetAPI } from './createRetweet.js';
export { default as deleteRetweetAPI } from './deleteRetweet.js';
