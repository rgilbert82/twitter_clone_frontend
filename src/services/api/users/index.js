export { default as getUsersAPI } from './getUsers.js';
export { default as getUserAPI } from './getUser.js';
export { default as createUserAPI } from './createUser.js';
export { default as updateUserAPI } from './updateUser.js';
export { default as getUserTweetsAPI } from './getUserTweets.js';
export { default as deleteUserAPI } from './deleteUser.js';
