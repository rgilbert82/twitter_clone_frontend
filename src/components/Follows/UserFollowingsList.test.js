jest.mock('../../services/api/follows');

import React from 'react';
import { shallow } from 'enzyme';
import { UserFollowingsList } from '.';
import { getUserFollowingsAPI } from '../../services/api/follows';

describe(UserFollowingsList, () => {
  let fetchSpy = jest.spyOn(UserFollowingsList.prototype, 'fetchFollowings');
  let component;

  beforeEach(() => {
    component = shallow(<UserFollowingsList user_id={1} loggedIn={true} />);
  });

  afterAll(() => {
    fetchSpy.mockClear();
  });

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("calls fetchFollowings on mount", () => {
    expect(fetchSpy).toHaveBeenCalled();
  });

  it("fetches the followers", () => {
    expect.assertions(1);
    component.instance().fetchFollowings()
      .then(() => {
        expect(component.state('followings')).toEqual([{id: 1, name: 'Ryan', username: 'RG', slug: 'rg'}]);
      });
  });
});
