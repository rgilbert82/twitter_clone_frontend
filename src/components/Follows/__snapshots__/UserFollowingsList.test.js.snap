// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class UserFollowingsList extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.state = { followings: [] };

    this.fetchFollowings = this.fetchFollowings.bind(this);
    this.unfollowUser = this.unfollowUser.bind(this);
  }

  componentDidMount() {
    this._isMounted = true;
    if (this._isMounted) {
      this.fetchFollowings();
    }
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  fetchFollowings() {
    return (0, _follows.getUserFollowingsAPI)(this.props.user_id).
    then(data => {
      this.setState({ followings: data });
    }).catch(() => {
      this.props.displayMessage('There was an error loading the followings');
    });
  }

  unfollowUser(userID) {
    return (0, _follows.deleteFollowAPI)(userID).
    then(() => {
      this.setState({ followings: this.state.followings.filter(fol => {return fol.id !== userID;}) });
      this.props.deleteFollowFromList();
    }).catch(() => {
      this.props.displayMessage('There was an error unfollowing the user');
    });
  }

  render() {
    const followingsList = this.state.followings.map(user => {
      return (
        _react2.default.createElement('li', { className: 'list_item', key: user.id, __source: { fileName: _jsxFileName, lineNumber: 48 }, __self: this },
          _react2.default.createElement(_Users.UserListItem, { user: user, isCurrentUserPage: this.props.isCurrentUserPage, unfollowUser: this.unfollowUser, __source: { fileName: _jsxFileName, lineNumber: 49 }, __self: this })));


    });

    return (
      _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 55 }, __self: this },
        _react2.default.createElement('h2', { className: 'sub_header', __source: { fileName: _jsxFileName, lineNumber: 56 }, __self: this }, 'Following'),
        _react2.default.createElement('ul', { __source: { fileName: _jsxFileName, lineNumber: 57 }, __self: this },
          followingsList)));



  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserFollowingsList
    loggedIn={true}
    user_id={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="sub_header"
>
          Following
</h2>,
        <ul>
          
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Following",
          "className": "sub_header",
        },
        "ref": null,
        "rendered": "Following",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="sub_header"
>
            Following
</h2>,
          <ul>
            
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Following",
            "className": "sub_header",
          },
          "ref": null,
          "rendered": "Following",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
